p2p/protocol/identify/obsaddr.go

```diff
@@ -418,6 +434,87 @@ func (oas *ObservedAddrManager) recordObservationUnlocked(conn network.Conn, obs
 	oas.addrs[localString] = append(oas.addrs[localString], oa)
 }
 
+// For a given transport Protocol (TCP/UDP):
+//
+// 1. If we have an activated address, we are behind an Cone NAT.
+// With regards to RFC 3489, this could be either a Full Cone NAT, a Restricted Cone NAT or a
+// Port Restricted Cone NAT. However, we do NOT differentiate between them here and simply classify all such NATs as a Cone NAT.
+//
+// 2. If four different peers observe a different address for us on outbound connections, we
+// are MOST probably behind a Symmetric NAT.
+//
+// Please see the documentation on the enumerations for `network.NATDeviceType` for more details about these NAT Device types
+// and how they relate to NAT traversal via Hole Punching.
+func (oas *ObservedAddrManager) emitNATTypeUnlocked() {
```

Comment:
why is this method named `Unlocked`?  maybe this is `observeAllNATTypes` and the next one is `observeSpecificNATType`? the naming of the two feels not quite representative of what they do.

Reply:
yeah what's with the name? Is there a locked variant?

Reply:
Have fixed this.